charts/prometheus/templates/sts.yaml

```
@@ -144,6 +144,9 @@ spec:
         {{- end }}
           ports:
             - containerPort: 9090
+          {{- if .Values.server.portName }}
+              portName: {{ .Values.server.portName }}
```

```suggestion
              name: {{ .Values.server.portName }}
```

The new field should be present in the values file.